modules/margin-protocol/src/lib.rs

```diff
@@ -1,41 +1,48 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use codec::{Decode, Encode};
-use frame_support::{decl_error, decl_event, decl_module, decl_storage};
+use codec::{Decode, Encode, FullCodec};
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, Parameter};
 use sp_arithmetic::Permill;
-use sp_runtime::{traits::StaticLookup, DispatchResult, RuntimeDebug};
+use sp_runtime::{
+	traits::{AtLeast32Bit, MaybeSerializeDeserialize, StaticLookup},
+	DispatchResult, RuntimeDebug,
+};
 // FIXME: `pallet/frame-` prefix should be used for all pallet modules, but currently `frame_system`
 // would cause compiling error in `decl_module!` and `construct_runtime!`
 // #3295 https://github.com/paritytech/substrate/issues/3295
 use frame_system as system;
 use frame_system::ensure_signed;
 use orml_traits::{MultiCurrency, PriceProvider};
 use orml_utilities::Fixed128;
-use primitives::{Balance, CurrencyId, Leverage, LiquidityPoolId, Price};
+use primitives::{Leverage, LiquidityPoolId, Price};
```

Comment:
As modules in laminar-chain are not libs for reuse, as discussed with @xlc , we would prefer to use concrete types from primitives module. This will makes impl easier by avoiding a lot of duplicated `Trait` types, type alias, and type conversions.

Pls refer to PR #91 for details, and keep types usage in margin protocol module in this PR consistent with current master branch.

Reply:
@shaopengw In that case we need to update it in other modules as well